tokio/src/sync/tests/loom_rwlock.rs

```diff
@@ -76,3 +119,39 @@ fn concurrent_read_write() {
         assert_eq!(10, *guard);
     });
 }
+
+#[test]
+fn concurrent_spin_read_write() {
+    let mut b = loom::model::Builder::new();
+
+    b.check(|| {
+        let rwlock = Arc::new(RwLock::<u32>::new(0));
+
+        let rwclone = rwlock.clone();
+        let t1 = thread::spawn(move || {
+            let mut guard = spin_write(&rwclone);
+            *guard += 5;
+        });
+
+        let rwclone = rwlock.clone();
+        let t2 = thread::spawn(move || {
+            let mut guard = spin_write(&rwclone);
+            *guard += 5;
+        });
+
+        let rwclone = rwlock.clone();
+        let t3 = thread::spawn(move || {
+            let guard = spin_read(&rwclone);
+            //at this state the value on the lock may either be 0, 5, or 10
```

Comment:
nit: prefer a space after the comment
```suggestion
            // at this state the value on the lock may either be 0, 5, or 10
```